### DISCUSSION TITLE
Mock matching log?

### DISCUSSION THREAD

**Discussion by tsfischer** [surpher/PactSwift#73] [Q&A] [answered]
Hi there, just started using this library today, and I'm attempting to convert over some code that has been using PactConsumerSwift.

The conversion has been pretty straightforward, however my tests don't work because the Mock Server isn't matching the request.

PactConsumerSwift was pretty good about explaining this right in an inline code error, basically saying "we expected these requests, but got these instead, here's a diff". Is there something similar with PactSwift? I'm currently at a loss as to why my responses aren't matching up.

EDIT: Turns out my problem was related to the Matcher stuff being handled differently than PactConsumerSwift. But I'd still be interested to know if there's a log file written out.

**Comment by surpher** [comment, marked as answer]
PactConsumerSwift is not really displaying the log, it is all `pact-ruby-standalone` that does it.

There's no log file written though. Can/could be put in the backlog to be done. `pact_ffi` library `PactSwiftMockServer` uses does expose a `const char *pactffi_fetch_log_buffer(const char *log_id);` and there's `int pactffi_log_to_file(const char *file_name, LevelFilter level_filter);`. Would need to expose the interface through `PactSwiftMockServer` and `PactSwift`.  Could be an optional `URL(pathToFile:)` for `MockService` initialiser.

Edit: https://github.com/surpher/PactSwift/issues/75

**Comment by surpher** [comment]
Though there is a `PACT_ENABLE_LOGGING` env var you can set to `all` for some verbosity in the debugger area:

![Screen Shot 2021-10-05 at 5 46 44 pm](https://user-images.githubusercontent.com/1733327/135973631-915a8c5b-c06a-4440-991f-d30522ef3beb.png)
